ipam.go

```diff
@@ -23,12 +23,20 @@ type Ipamer interface {
 	AcquireSpecificIP(prefixCidr, specificIP string) (*IP, error)
 	// AcquireIP will return the next unused IP from this Prefix.
 	AcquireIP(prefixCidr string) (*IP, error)
+	// Acquire will attempt to acquire specified IPs by provided arguments.
+	// specificIPs shows how many specific IPs to return. In case any of them overlaps an ErrAlreadyAllocated is returned.
+	// dynamicIPs shows how many of any available IPs to return.
+	// If there is no free IP an NoIPAvailableError is returned.
+	Acquire(prefixCidr string, specificIPs []string, dynamicIPs uint64) ([]*IP, error)
```

Comment:
I need to allocate many IPs under heavy load. Main idea is to manage more IPs in less DB interactions.

Problem was that regular Acquire/ReleaseIP functions triggers Storage transaction on every time.
```
for i := 0; i < 500; i++ {
	go func() {
		for {
			ip, err := ipam.AcquireIP()
			if err != nil {
				fmt.Printf("AcquireIP: %v\n", err)
				continue
			}
			err = ipam.ReleaseIP(ip)
			if err != nil {
				fmt.Printf("ReleaseIP: %v\n", err)
			}
		}
	}()
}
```
Due to optimistic lock and 10 retries there will be many errors in about 1 minute. 

Plus such heavy Storage usage requires many TCP transaction resulting network congestion:
1. Read prefix to allocate next free IP - 1 GET command TCP transaction
2. Start transaction - 1 WATCH command TCP transaction
3. Read prefix one more time to check "version" - 1 GET command TCP transaction
4. Pipelined set and commit transaction - SET + EXEC commands in 1 TCP transaction

On transaction collision (optimistic lock fail) all this must be redone one more time. And on high load chance of collision raises very quickly. Additionally go routines wait long time because of network response latency.

Reply:
OK sure, on simpler approach would be to have one Acquire for specific and one for dynamic IPs. But still, what happens if not all of the requested IPs could be acquired, are none of them acquired and a error for the ip which could not be acquired is returned. Very strange interface from my POV.

Reply:
What performance requirements do you have which could not be met with the current redis implementation ?

Reply:
Sure, I can rework interface. This was done like that so minimize interface changes and just make working solution in fork.

As for errors, it is the same safe mechanism as in master for single IP operation - optimistic lock.
```
func (i *ipamer) Acquire(prefixCidr string, specificIPs []string, dynamicIPs uint64) ([]*IP, error) {
	var ips []*IP
	return ips, retryOnOptimisticLock(func() error {
		var err error
		ips, err = i.acquireIPInternal(prefixCidr, specificIPs, dynamicIPs)
		return err
	})
}
```
On errors, or on IP shortage the whole operation will be re-done from scratch. That is 
1. PrefixFrom() - will re-read fresh CIDR status
2. for ip := ipnet.Range().From(); ...; ip = ip.Next() - will iterate all available IPs
3. storage.UpdatePrefix() - will attempt to write new Prefix state.

All will be done `retry.Attempts(10)` times as it is configured now

Same for new Release() operation.

As for performance, problem is that we need to process several thousand IP allocations/sec using _external Redis_. This is distributed solution thus Redis also serves other instances of same application. Plus we must use lib in conjunction with other Redis keys. So as mentioned above _4 TCP transactions for 1 IP_ creates congestion and overloads Redis server. Allocating let's say 100 IPs per Redis transaction is much faster.

Reply:
Why do you need to reuse a existing redis installation ? I started work on exposing this library with an grpc server endpoint #46 , then you will be able to use this instance of go-ipam which uses one single purpose redis instance which will for sure perform with several thousands of allocs/sec

Reply:
BTW, gRPC service integration is actually a good idea in general. This will certainly improve dedicated server usage.